rllib/core/rl_module/rl_module.py

```diff
@@ -244,6 +249,36 @@ def get_initial_state(self) -> NestedDict:
         """
         return {}
 
+    @property
+    def view_requirements(self) -> Mapping[str, ViewRequirement]:
+        """Returns the view requirements of the module."""
+        vr = self.__get_default_view_requirements()
+
+        # get the initial state in numpy format, infer the state from it, and create
+        # an apporpriate view requirement.
+        init_state = convert_to_numpy(self.get_initial_state())
+        if init_state:
+            init_state = tree.map_structure(lambda x: x[None], init_state)
+            space = get_gym_space_from_struct_of_tensors(init_state, batched_input=True)
+            max_seq_len = self.config.max_seq_len
+            assert max_seq_len is not None
```

Comment:
I put an assert here now, but this is very brittle.
I put max_seq_len=1 as a default in the RLModuleConfig now. It was None before and that leads to weird errors in the agent collector.